Loop/Managers/AlertMuter.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  AlertMuter.swift
+//  Loop
+//
+//  Created by Nathaniel Hamming on 2022-09-14.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftUI
+import LoopKit
+
+public class AlertMuter: ObservableObject {
+    struct Configuration: Equatable, RawRepresentable {
+        typealias RawValue = [String: Any]
+
+        enum ConfigurationKey: String {
+            case enabled
+            case duration
+            case startTime
+        }
+
+        init?(rawValue: [String : Any]) {
+            guard let enabled = rawValue[ConfigurationKey.enabled.rawValue] as? Bool
+            else { return nil }
+
+            self.enabled = enabled
+            self.duration = rawValue[ConfigurationKey.enabled.rawValue] as? TimeInterval
+            self.startTime = rawValue[ConfigurationKey.startTime.rawValue] as? Date
+        }
+
+        var rawValue: [String : Any] {
+            var rawValue: [String : Any] = [:]
+            rawValue[ConfigurationKey.enabled.rawValue] = enabled
+            rawValue[ConfigurationKey.duration.rawValue] = duration
+            rawValue[ConfigurationKey.startTime.rawValue] = startTime
+            return rawValue
+        }
+
+        var enabled: Bool {
+            didSet {
+                guard enabled else { return }
+                startTime = Date()
+            }
+        }
+        var duration: TimeInterval?
+        private(set) var startTime: Date?
+
+        init(enabled: Bool, duration: TimeInterval?) {
+            self.enabled = enabled
+            self.duration = enabled ? duration : nil
+            self.startTime = enabled ? Date() : nil
+        }
+
+        var shouldMuteAlerts: Bool {
+            shouldMuteAlert()
+        }
+
+        func shouldMuteAlert(_ fromNow: TimeInterval = 0) -> Bool {
```

Comment:
Suggest `shouldMuteAlert(at alertDate: Date)`, as the caller is doing `timeIntervalSinceNow` on the actual alert date, and you're unwinding that here.  Simpler (and more intuitive imo), to just use the alertDate.